### PR TITLE
fixed 569-app crash on invalid input in split fragment

### LINKED ISSUE #569: App crashes when invalid input is entered into Split PDF Fragment
<a href="url"></a>
The Edit Text  in Split Fragment takes in the input in **1,2-3,5-7,8** format which then splits the pdf accordingly.    
When the user enters anything like alphabets,special characters etc the App crashes.The edittext input needs to be validated and user should be informed about incorrect input.

### PR DESCRIPTION
**Pull request by Ni3verma**
# Description
app was crashing when any letter was inserted. I fixed the issue by allowing the user to enter only digits and comma and dash sign

Fixes #569 

## Type of change
Just put an x in the [] which are valid.
- [x ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes.
- [ x] `./gradlew assembleDebug assembleRelease`
- [ x] `./gradlew checkstyle`

# Checklist:
- [x ] My code follows the style guidelines of this project
- [x ] I have performed a self-review of my own code
- [x ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x ] My changes generate no new warnings

### CODE REVIEW

**Comment by srv-twry** [comment]
@Ni3verma I don't think it is possible to handle all the cases using XML. I would suggest adding a method that checks the user input after the `SPLIT PDF` button is clicked. That method should handle all the cases one by one. In case of an error, show appropriate error messages to the user.

**Comment by Swati4star** [comment]
@Ni3verma updates?

**Comment by Ni3verma** [comment]
@Swati4star , In **Progress** - will push code by tomorrow..

**Comment by Swati4star** [comment]
@srv-twry Review this PR.

File: app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java [hunk 1]
```diff
@@ -120,6 +119,9 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) throw
 
     @OnClick(R.id.splitFiles)
     public void parse() {
+        // hide the soft keyboard on click of this button
+        hideKeyboard(mActivity);
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java]: I would suggest removing this comment. The functionality is pretty clear just by the name of the method called. Comment is unnecessary.

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java [hunk 2]
```diff
@@ -574,6 +578,39 @@ public boolean reorderRemovePDF(String inputPath, String output, String pages) {
...
     }
 
+    private boolean isInputValid(String path, String[] ranges) {
+        try {
+            PdfReader reader = new PdfReader(path);
+            int numOfPages = reader.getNumberOfPages();
+            int startPage;
+            int endPage;
+
+
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: Remove extra line.

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java [hunk 3]
```diff
@@ -574,6 +578,39 @@ public boolean reorderRemovePDF(String inputPath, String output, String pages) {
...
+                    endPage = Integer.parseInt(range.substring(range.indexOf("-") + 1));
+                    if (startPage > numOfPages || endPage > numOfPages) {
+                        showSnackbar(mContext, R.string.error_page_number);
+                        return false;
+                    } else if (startPage >= endPage) {
+                        showSnackbar(mContext, R.string.error_page_range);
+                        return false;
+                    }
+                }
+            }
```
Comment (srv-twry) [inline app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: From dry running on example `1-4 2-3`, I think this method will consider it valid but is it actually valid?

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java (hunk 2 above)
Comment (Ni3verma) [reply app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: ok

File: app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java (hunk 1 above)
Comment (Ni3verma) [reply app/src/main/java/swati4star/createpdf/fragment/SplitFilesFragment.java]: ok

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java (hunk 3 above)
Comment (Ni3verma) [reply app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: why not? I think it's upto user to set how many pages he want in a single pdf.
1-4, 2-3 is a valid input.

**Comment by Ni3verma** [comment]
> This is one of those cases where writing unit test is the best way to ensure that all the cases are handled properly. Please add unit-test for the `isInputValid` by checking it on various edge cases that you could think of. This would ensure that any future changes won't break the application.

I have never added unit tests. can you please explain where should I write this code? any links to learn more about it?

File: app/src/main/java/swati4star/createpdf/util/PDFUtils.java (hunk 3 above)
Comment (Swati4star) [reply app/src/main/java/swati4star/createpdf/util/PDFUtils.java]: I think the input seems valid :+1:
You can leave it as is.

**Comment by Swati4star** [comment]
We can create a different issue for adding unit tests for this functions, as this issue is an easy one.

@srv-twry can you create a different issue for adding unit tests?
